backend/execution_cache/src/mutation.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ArgoWorkflowTemplate string = "workflows.argoproj.io/template"
+	ExecutionKey         string = "pipelines.kubeflow.org/execution_cache_key"
+	AnnotationPath       string = "/metadata/annotations"
+)
+
+const (
+	Add OperationType = "add"
+)
+
+var (
+	podResource = metav1.GroupVersionResource{Version: "v1", Resource: "pods"}
+)
+
+// mutatePodIfCached will check whether the execution has already been run before from MLMD and apply the output into pod.metadata.output
+func mutatePodIfCached(req *v1beta1.AdmissionRequest) ([]patchOperation, error) {
+	// This handler should only get called on Pod objects as per the MutatingWebhookConfiguration in the YAML file.
+	// However, if (for whatever reason) this gets invoked on an object of a different kind, issue a log message but
+	// let the object request pass through otherwise.
+	if req.Resource != podResource {
+		log.Printf("expect resource to be %q", podResource)
+		return nil, nil
+	}
+
+	// Parse the Pod object.
+	raw := req.Object.Raw
+	pod := corev1.Pod{}
+	if _, _, err := universalDeserializer.Decode(raw, nil, &pod); err != nil {
+		return nil, fmt.Errorf("could not deserialize pod object: %v", err)
+	}
+
+	var patches []patchOperation
+	annotations := pod.ObjectMeta.Annotations
+	template, exists := annotations[ArgoWorkflowTemplate]
```

Comment:
Same here, I'd vote for returning fast
```go
if !exists {
        return patches, nil
}
```
Then the rest won't require `if`s and such.